pkg/controller/nodes/branch/evaluator.go

```diff
@@ -24,10 +24,14 @@ func EvaluateComparison(expr *core.ComparisonExpression, nodeInputs *core.Litera
 	var rPrim *core.Primitive
 
 	if expr.GetLeftValue().GetPrimitive() == nil {
-		if nodeInputs == nil {
-			return false, errors.Errorf(ErrorCodeMalformedBranch, "Failed to find Value for Variable [%v]", expr.GetLeftValue().GetVar())
+		if len(expr.GetLeftValue().GetVar()) == 0 {
```

Comment:
So we're representing a `None` value by using a variable of length 0? and then automatically converting it into a scalar with `NoneValue`. Would it make more sense to set this as a scalar with `NoneValue` from flytekit?

Reply:
> Would it make more sense to set this as a scalar with NoneValue from flytekit?

Yes, we can, however, the idl must be changed since none is not in the primitives. we can add `is_none bool` to the [operand](https://github.com/flyteorg/flyteidl/blob/95e11cca2dac18b727f122bbc4456ea6ab499289/protos/flyteidl/core/condition.proto#L30), wdyt?

Reply:
Is your intuition that we have `Primitive` rather than `Scalar` in the operand because this was meant to be very simple?

I think I would support adding a `Void none_type` to the operand [similar to this](https://github.com/flyteorg/flyteidl/blob/95e11cca2dac18b727f122bbc4456ea6ab499289/protos/flyteidl/core/literals.proto#L86).

Reply:
updated Idl. https://github.com/flyteorg/flyteidl/pull/427